### PR TITLE
add anomaly charts to chartcuterie

### PR DESCRIPTION
**Pull request by nhsiehgit**
Screenshot 2024-09-05 at 3 52 14 PM

we now have anomaly lines rendering in chartcuterie charts!

follow up to https://github.com/getsentry/sentry/pull/76889

### CODE REVIEW

File: src/sentry/web/frontend/debug/debug_chart_renderer.py [hunk 1]
```diff
@@ -524,35 +704,133 @@ def get(self, request: HttpRequest) -> HttpResponse:
         ret = []
 
         ret.append(
-            charts.generate_chart(ChartType.SLACK_DISCOVER_TOTAL_PERIOD, discover_total_period)
+            {
```
Comment (nhsiehgit) [inline src/sentry/web/frontend/debug/debug_chart_renderer.py:707]: This refactor is mainly to make it easier to see what chart is what in the debug view

File: src/sentry/templates/sentry/debug/chart-renderer.html [hunk 2]
```diff
@@ -22,7 +22,8 @@
     <h2>Chart Renderer Debug</h2>
     <div class="charts">
       {% for chart_url in charts %}
-        <img src="{{ chart_url }}" />
+        <div>{{ chart_url.title }}</div>
```
Comment (nhsiehgit) [inline src/sentry/templates/sentry/debug/chart-renderer.html:25]: adds title to chart in the debug view

File: src/sentry/web/debug_urls.py [hunk 3]
```diff
@@ -148,12 +148,12 @@
     re_path(r"^debug/mail/sso-linked/$", DebugSsoLinkedEmailView.as_view()),
     re_path(r"^debug/mail/sso-unlinked/$", DebugSsoUnlinkedEmailView.as_view()),
     re_path(
-        r"^debug/mail/sso-unlinked/no-password$", DebugSsoUnlinkedNoPasswordEmailView.as_view()
+        r"^debug/mail/sso-unlinked/no-password/$", DebugSsoUnlinkedNoPasswordEmailView.as_view()
```
Comment (nhsiehgit) [inline src/sentry/web/debug_urls.py:151]: missing trailing slashes in debug routes

File: src/sentry/web/frontend/debug/debug_chart_renderer.py [hunk 4]
```diff
@@ -524,35 +704,133 @@ def get(self, request: HttpRequest) -> HttpResponse:
...
+        )
+        ret.append(
+            {
+                "chart": charts.generate_chart(
+                    ChartType.SLACK_METRIC_ALERT_SESSIONS, crash_free_metric_alert
+                ),
+                "title": "slack metric alert sessions crash free",
+            }
+        )
+        ret.append(
```
Comment (nhsiehgit) [inline src/sentry/web/frontend/debug/debug_chart_renderer.py:827]: This is the new chart of error events _with_ anomalies

File: src/sentry/incidents/charts.py [hunk 5]
```diff
@@ -203,6 +228,16 @@ def build_metric_alert_chart(
             user,
         ),
     }
+    if features.has(
```
Comment (ceorourke) [inline src/sentry/incidents/charts.py:231]: shouldn't this check if the `detection_type` on the `alert_rule` is dynamic as well?

File: src/sentry/web/frontend/debug/debug_chart_renderer.py [hunk 6]
```diff
@@ -447,7 +447,185 @@
     "incidents": [incident],
     "selectedIncident": None,
 }
-
+metric_alert_with_anomalies = metric_alert.copy()
+metric_alert_with_anomalies["anomalies"] = [
+    {
```
Comment (ceorourke) [inline src/sentry/web/frontend/debug/debug_chart_renderer.py:452]: it's just a debug view so it's not critical but you can use these types https://github.com/getsentry/sentry/blob/master/src/sentry/seer/anomaly_detection/types.py#L5-L13

Comment (ceorourke) [reply src/sentry/web/frontend/debug/debug_chart_renderer.py:452]: actually I think `anomaly_type` in the types should be of type `AnomalyType`

File: src/sentry/web/frontend/debug/debug_chart_renderer.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/web/frontend/debug/debug_chart_renderer.py:705]: nittiest nit but can you make Discover capitalized? it makes it clearer it's the product Discover and not just ... Slack discovering something

File: src/sentry/incidents/charts.py (hunk 5 above)
Comment (nhsiehgit) [reply src/sentry/incidents/charts.py:234]: Yes you are correct :O

File: src/sentry/web/frontend/debug/debug_chart_renderer.py (hunk 6 above)
Comment (nhsiehgit) [reply src/sentry/web/frontend/debug/debug_chart_renderer.py:452]: Oh oof.
this must've gotten lost in rebase.

will fix this

**Review by mifu67 — APPROVED** [review]
lgtm (???)

**Pull request merged**